app/storage/s3ResourceHandler.js

```diff
@@ -19,91 +19,186 @@
 const conf = require('./../conf.js').conf;
 const S3ClientClass = require('./s3NewClient');
 const cipher = require('./cipher');
+const { v4: uuid } = require('uuid');
+const { models } = require('../apollo/models');
+const _ = require('lodash');
 
 class S3ResourceHandler {
+  logger = null;
 
-  constructor(logger, resourceKey, bucketName, location, endpoint) {
-    this.logger = logger;
-    if (!resourceKey || !bucketName) {
-      throw new Error(`Path (${resourceKey}) and/or bucket name (${bucketName}) is not specified`);
+  constructor(args){
+    var { logger, path, bucketConfObj, endpoint, org } = args;
```

Comment:
Use `const` instead of `var` in this and many other locations in the PR
```suggestion
    const { logger, path, bucketConfObj, endpoint, org } = args;
```